src/librustc_span/lib.rs

```diff
@@ -883,6 +889,101 @@ impl ExternalSource {
 #[derive(Debug)]
 pub struct OffsetOverflowError;
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
+pub enum SourceFileChecksumKind {
+    Stable,
+    Md5,
```

Comment:
If we do this, I suppose we can just remove `Stable` eventually, and let `Md5` be the default.